ibu-imager/seedcleanup/seedcleanup.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package seedcleanup
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/openshift-kni/lifecycle-agent/ibu-imager/common"
+	"github.com/openshift-kni/lifecycle-agent/ibu-imager/ops"
+	"github.com/openshift-kni/lifecycle-agent/internal/clusterconfig"
+	"github.com/sirupsen/logrus"
+)
+
+// SeedCleaner handles cleanup operations after creating a seed image, removing temporary files
+// and executing additional cleanup steps as needed.
+type SeedCleaner struct {
+	log                  *logrus.Logger
+	ops                  ops.Ops
+	backupDir            string
+	containerRegistry    string
+	authFile             string
+	recertContainerImage string
+	recertSkipValidation bool
+}
+
+func NewSeedCleaner(log *logrus.Logger, ops ops.Ops, backupDir,
+	containerRegistry, authFile, recertContainerImage string, recertSkipValidation bool) *SeedCleaner {
+
+	return &SeedCleaner{
+		log:                  log,
+		ops:                  ops,
+		backupDir:            backupDir,
+		containerRegistry:    containerRegistry,
+		authFile:             authFile,
+		recertContainerImage: recertContainerImage,
+		recertSkipValidation: recertSkipValidation,
+	}
+}
+
+// CleanupSeedCluster comprises the Imager workflow for cleanup operations after creating a seed image
+// out of an SNO cluster.
+func (s *SeedCleaner) CleanupSeedCluster() error {
+	s.log.Println("Cleaning up seed cluster")
+
+	if err := s.cleanupBackupDir(); err != nil {
+		return err
+	}
+
+	if err := s.cleanupConfigurationFiles(); err != nil {
+		return err
+	}
+
+	if s.recertSkipValidation {
+		s.log.Info("Skipping restoring certificates in etcd via recert tool.")
+	} else {
+		if err := s.restoreCertificates(); err != nil {
+			return err
+		}
+	}
+
+	if err := s.restoreCluster(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *SeedCleaner) cleanupBackupDir() error {
+	if err := os.RemoveAll(s.backupDir); err != nil {
+		return fmt.Errorf("error removing %s: %w", s.backupDir, err)
+	}
+
+	if err := os.Remove(common.BackupChecksDir + "/container_list.done"); err != nil && !os.IsNotExist(err) {
```

Comment:
What is the reason we're only deleting `container_list.done` in particular, and not the entire checks dir?